{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import sqlite3\n",
    "from snps import SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfs = {}\n",
    "for i in os.listdir(\"ADHD\"):\n",
    "    if i in [\".DS_store\", \".ipynb_checkpoints\"]:\n",
    "        continue\n",
    "    try:\n",
    "        s = SNPs(f\"ADHD/{i}\")\n",
    "    except UnicodeDecodeError:\n",
    "        continue\n",
    "    dfs[i] = s.snps\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Position</th>\n",
       "      <th>Ref_Base</th>\n",
       "      <th>Alt_Base</th>\n",
       "      <th>Note</th>\n",
       "      <th>Coding</th>\n",
       "      <th>Hugo</th>\n",
       "      <th>Sequence_Ontology</th>\n",
       "      <th>Protein_Change</th>\n",
       "      <th>Samples</th>\n",
       "      <th>Chrom.1</th>\n",
       "      <th>Position.1</th>\n",
       "      <th>Disease_Names</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Global_AF</th>\n",
       "      <th>Chemical</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>43708410</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ST3GAL3</td>\n",
       "      <td>intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60.23andme.27;616.23andme.288;8.23andme.2;865....</td>\n",
       "      <td>chr1</td>\n",
       "      <td>43946668</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.435</td>\n",
       "      <td>NaN</td>\n",
       "      <td>het;hom;het;het</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>43712399</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ST3GAL3</td>\n",
       "      <td>intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60.23andme.27;616.23andme.288;8.23andme.2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>43950657</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.784</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hom;het;het</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>43729682</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ST3GAL3</td>\n",
       "      <td>intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60.23andme.27;616.23andme.288;8.23andme.2;865....</td>\n",
       "      <td>chr1</td>\n",
       "      <td>43967940</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hom;hom;het;het</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>43734985</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ST3GAL3</td>\n",
       "      <td>intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60.23andme.27;616.23andme.288;8.23andme.2;865....</td>\n",
       "      <td>chr1</td>\n",
       "      <td>43973243</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>het;het;het;het</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>43734985</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ST3GAL3</td>\n",
       "      <td>intron</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60.23andme.27;616.23andme.288;8.23andme.2;865....</td>\n",
       "      <td>chr1</td>\n",
       "      <td>43973243</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>het;het;het;het</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chrom  Position Ref_Base Alt_Base  Note Coding     Hugo Sequence_Ontology  \\\n",
       "0  chr1  43708410        G        A   NaN    NaN  ST3GAL3            intron   \n",
       "1  chr1  43712399        T        G   NaN    NaN  ST3GAL3            intron   \n",
       "2  chr1  43729682        G        C   NaN    NaN  ST3GAL3            intron   \n",
       "3  chr1  43734985        C        A   NaN    NaN  ST3GAL3            intron   \n",
       "4  chr1  43734985        C        G   NaN    NaN  ST3GAL3            intron   \n",
       "\n",
       "  Protein_Change                                            Samples Chrom.1  \\\n",
       "0            NaN  60.23andme.27;616.23andme.288;8.23andme.2;865....    chr1   \n",
       "1            NaN          60.23andme.27;616.23andme.288;8.23andme.2    chr1   \n",
       "2            NaN  60.23andme.27;616.23andme.288;8.23andme.2;865....    chr1   \n",
       "3            NaN  60.23andme.27;616.23andme.288;8.23andme.2;865....    chr1   \n",
       "4            NaN  60.23andme.27;616.23andme.288;8.23andme.2;865....    chr1   \n",
       "\n",
       "   Position.1 Disease_Names  Phenotype  Global_AF  Chemical         Zygosity  \n",
       "0    43946668           NaN        NaN      0.435       NaN  het;hom;het;het  \n",
       "1    43950657           NaN        NaN      0.784       NaN      hom;het;het  \n",
       "2    43967940           NaN        NaN        NaN       NaN  hom;hom;het;het  \n",
       "3    43973243           NaN        NaN        NaN       NaN  het;het;het;het  \n",
       "4    43973243           NaN        NaN        NaN       NaN  het;het;het;het  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FILE = \"db/ADHD_variant_genes.tsv\"\n",
    "df = pd.read_csv(FILE, sep='\\t')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "bigdf = pd.concat(dfs.values(), keys=dfs.keys())\n",
    "rsids = \"\"\"rs11420276\n",
    "rs1222063\n",
    "rs4858241\n",
    "rs28411770\n",
    "rs4916723\n",
    "rs5886709\n",
    "rs74760947\n",
    "rs11591402\n",
    "rs1427829\n",
    "rs281324\n",
    "rs212178\"\"\".split(\"\\n\")\n",
    "keys = list(dfs.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chrom              5\n",
      "pos         87854395\n",
      "genotype          CC\n",
      "Name: (678.23andme.txt, rs4916723), dtype: object\n",
      "chrom              5\n",
      "pos         87854395\n",
      "genotype          CC\n",
      "Name: (561.23andme.txt, rs4916723), dtype: object\n",
      "chrom              5\n",
      "pos         87854395\n",
      "genotype          CC\n",
      "Name: (genome_Susan_Loftus_v3_Full_20200212185613 2.txt, rs4916723), dtype: object\n",
      "chrom              5\n",
      "pos         87890151\n",
      "genotype          AC\n",
      "Name: (8.23andme.txt, rs4916723), dtype: object\n",
      "chrom              5\n",
      "pos         87890151\n",
      "genotype          AA\n",
      "Name: (60.23andme.txt, rs4916723), dtype: object\n",
      "chrom              5\n",
      "pos         87854395\n",
      "genotype          AC\n",
      "Name: (genome_Emma_Loftus_v3_Full_20200324162732.txt, rs4916723), dtype: object\n",
      "chrom              5\n",
      "pos         87890151\n",
      "genotype          AA\n",
      "Name: (865.23andme.txt, rs4916723), dtype: object\n",
      "chrom              5\n",
      "pos         87890151\n",
      "genotype          AC\n",
      "Name: (616.23andme.txt, rs4916723), dtype: object\n"
     ]
    }
   ],
   "source": [
    "c = defaultdict(int)\n",
    "for j in keys:\n",
    "    for i in rsids:\n",
    "        try:\n",
    "            c[i] += 1\n",
    "            print(bigdf.loc[j, i])\n",
    "        except KeyError:\n",
    "            c[i] = 0\n",
    "            continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(int,\n",
       "            {'rs11420276': 0,\n",
       "             'rs1222063': 0,\n",
       "             'rs4858241': 0,\n",
       "             'rs28411770': 0,\n",
       "             'rs4916723': 4,\n",
       "             'rs5886709': 0,\n",
       "             'rs74760947': 0,\n",
       "             'rs11591402': 0,\n",
       "             'rs1427829': 0,\n",
       "             'rs281324': 0,\n",
       "             'rs212178': 0})"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = \"\"\"ST3GAL3\n",
    "SPAG16\n",
    "PCDH7\n",
    "LINC00461\n",
    "FOXP2\n",
    "LINC01288\n",
    "SORCS3\n",
    "DUSP6\n",
    "SEMA6D\n",
    "DRD4\n",
    "DAT1\n",
    "DBH\n",
    "MAOA\n",
    "DRD5\"\"\".split(\"\\n\")\n",
    "# df[df[\"Hugo\"] == \"FOXP2\"]\n",
    "df[df[\"Hugo\"] == \"FOXP2\"].Samples.str.split(\";\", expand=True)\n",
    "gdfs = {g: df[df[\"Hugo\"] == g] for g in genes}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['ST3GAL3', 'SPAG16', 'PCDH7', 'LINC00461', 'FOXP2', 'LINC01288', 'SORCS3', 'DUSP6', 'SEMA6D', 'DRD4', 'DAT1', 'DBH', 'MAOA', 'DRD5'])"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gdfs.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [Chrom, Position, Ref_Base, Alt_Base, Note, Coding, Hugo, Sequence_Ontology, Protein_Change, Samples, Chrom.1, Position.1, Disease_Names, Phenotype, Global_AF, Chemical, Zygosity]\n",
      "Index: []\n"
     ]
    }
   ],
   "source": [
    "for k, v in gdfs.items():\n",
    "    _df = v[gdfs[k].Chrom == \"chr7\"]\n",
    "    print(_df[(_df['Position.1'] >= 114446081) & (_df['Position.1'] <= 114446081)])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### variants in coding regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [],
   "source": [
    "coding = df[df.Coding.notna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "58"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(coding)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Position</th>\n",
       "      <th>Ref_Base</th>\n",
       "      <th>Alt_Base</th>\n",
       "      <th>Note</th>\n",
       "      <th>Coding</th>\n",
       "      <th>Sequence_Ontology</th>\n",
       "      <th>Protein_Change</th>\n",
       "      <th>Samples</th>\n",
       "      <th>Chrom.1</th>\n",
       "      <th>Position.1</th>\n",
       "      <th>Disease_Names</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Global_AF</th>\n",
       "      <th>Chemical</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hugo</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DBH</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAOA</th>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRD4</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DUSP6</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEMA6D</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPAG16</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SORCS3</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DRD5</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PCDH7</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ST3GAL3</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FOXP2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chrom  Position  Ref_Base  Alt_Base  Note  Coding  Sequence_Ontology  \\\n",
       "Hugo                                                                            \n",
       "DBH         13        13        13        13     0      13                 13   \n",
       "MAOA        11        11        11        11     0      11                 11   \n",
       "DRD4         8         8         8         8     0       8                  8   \n",
       "DUSP6        6         6         6         6     0       6                  6   \n",
       "SEMA6D       6         6         6         6     0       6                  6   \n",
       "SPAG16       4         4         4         4     0       4                  4   \n",
       "SORCS3       3         3         3         3     0       3                  3   \n",
       "DRD5         2         2         2         2     0       2                  2   \n",
       "PCDH7        2         2         2         2     0       2                  2   \n",
       "ST3GAL3      2         2         2         2     0       2                  2   \n",
       "FOXP2        1         1         1         1     0       1                  1   \n",
       "\n",
       "         Protein_Change  Samples  Chrom.1  Position.1  Disease_Names  \\\n",
       "Hugo                                                                   \n",
       "DBH                  13       13       13          13              1   \n",
       "MAOA                 11       11       11          11              0   \n",
       "DRD4                  8        8        8           8              0   \n",
       "DUSP6                 6        6        6           6              0   \n",
       "SEMA6D                6        6        6           6              0   \n",
       "SPAG16                4        4        4           4              0   \n",
       "SORCS3                3        3        3           3              0   \n",
       "DRD5                  2        2        2           2              0   \n",
       "PCDH7                 2        2        2           2              0   \n",
       "ST3GAL3               2        2        2           2              0   \n",
       "FOXP2                 1        1        1           1              0   \n",
       "\n",
       "         Phenotype  Global_AF  Chemical  Zygosity  \n",
       "Hugo                                               \n",
       "DBH              0          1         0        13  \n",
       "MAOA             0          0         0        11  \n",
       "DRD4             0          1         0         8  \n",
       "DUSP6            0          0         0         6  \n",
       "SEMA6D           0          1         0         6  \n",
       "SPAG16           0          0         0         4  \n",
       "SORCS3           0          0         0         3  \n",
       "DRD5             0          0         0         2  \n",
       "PCDH7            0          0         0         2  \n",
       "ST3GAL3          0          0         0         2  \n",
       "FOXP2            0          0         0         1  "
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_df = coding.groupby(\"Hugo\").count()\n",
    "count_df.sort_values(by=\"Coding\", ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.6.9 64-bit ('bioinf': venv)",
   "language": "python",
   "name": "python36964bitbioinfvenv6f83f1e357ac4c289c6588591627fd06"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
